src/EditorFeatures/CSharpTest/ConvertNamespace/ConvertNamespaceCommandHandlerTests.cs

```diff
@@ -0,0 +1,277 @@
+@"namespace N$$
+{
+    class C
+    {
+    }
+}");
+
+            testState.SendTypeChar(';');
+            testState.AssertCodeIs(
+@"namespace N;
+
+class C
+{
+}
+");
+        }
+
+        [WpfFact]
+        public void TestDottedName()
+        {
+            using var testState = ConvertNamespaceTestState.CreateTestState(
+@"namespace A.B$$
+{
+    class C
+    {
+    }
+}");
+
+            testState.SendTypeChar(';');
+            testState.AssertCodeIs(
+@"namespace A.B;
+
+class C
+{
+}
+");
+        }
+
+        [WpfFact]
+        public void TestAfterWhitespace()
+        {
+            using var testState = ConvertNamespaceTestState.CreateTestState(
+@"namespace A.B  $$
+{
+    class C
+    {
+    }
+}");
+
+            testState.SendTypeChar(';');
+            testState.AssertCodeIs(
+@"namespace A.B;
+
+class C
+{
+}
+");
+        }
+
+        [WpfFact]
+        public void TestBeforeName()
+        {
+            using var testState = ConvertNamespaceTestState.CreateTestState(
+@"namespace $$N
+{
+    class C
+    {
+    }
+}");
+
+            testState.SendTypeChar(';');
+            testState.AssertCodeIs(
+@"namespace ;N
+{
+    class C
+    {
+    }
+}");
+        }
+
+        [WpfFact]
+        public void TestNestedNamespace()
+        {
+            using var testState = ConvertNamespaceTestState.CreateTestState(
+@"namespace N$$
+{
+    namespace N2
+    {
+        class C
+        {
+        }
+    }
+}");
+
+            testState.SendTypeChar(';');
+            testState.AssertCodeIs(
+@"namespace N;
+{
+    namespace N2
+    {
+        class C
+        {
+        }
+    }
+}");
+        }
+
+        [WpfFact]
+        public void TestSiblingNamespace()
+        {
+            using var testState = ConvertNamespaceTestState.CreateTestState(
+@"namespace N$$
+{
+}
+
+namespace N2
+{
+    class C
+    {
+    }
+}");
+
+            testState.SendTypeChar(';');
+            testState.AssertCodeIs(
+@"namespace N;
+{
+}
+
+namespace N2
+{
+    class C
+    {
+    }
+}");
+        }
+
+        [WpfFact]
+        public void TestOuterUsings()
+        {
+            using var testState = ConvertNamespaceTestState.CreateTestState(
+@"
+using A;
+using B;
+
+namespace N$$
+{
+    class C
+    {
+    }
+}");
+
+            testState.SendTypeChar(';');
+            testState.AssertCodeIs(
+@"
+using A;
+using B;
+
+namespace N;
+
+class C
+{
+}
+");
+        }
+
+        [WpfFact]
+        public void TestInnerUsings()
+        {
+            using var testState = ConvertNamespaceTestState.CreateTestState(
+@"
+namespace N$$
+{
+    using A;
+    using B;
+
+    class C
+    {
+    }
+}");
+
+            testState.SendTypeChar(';');
+            testState.AssertCodeIs(
+@"
+namespace N;
+
+using A;
+using B;
+
+class C
+{
+}
+");
+        }
+
+        [WpfFact]
+        public void TestCommentAfterName()
+        {
+            using var testState = ConvertNamespaceTestState.CreateTestState(
+@"namespace N$$ // Goo
```

Comment:
This makes me wonder will happen for `namespace A$$.B`

Reply:
happyh to add a test.